src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectManaged.cs

```diff
@@ -166,10 +166,13 @@ internal override object ReadAsync(out uint error, ref object handle)
 
         internal override object CreateAndSetAttentionPacket()
         {
-            SNIPacket attnPacket = new SNIPacket(Handle);
-            _sniAsyncAttnPacket = attnPacket;
-            SetPacketData(attnPacket, SQL.AttentionHeader, TdsEnums.HEADER_LEN);
-            return attnPacket;
+            if (_sniAsyncAttnPacket == null)
```

Comment:
Is there any reason to remove the local `attnPacket` ? It is always better to work with a local before putting a reference on a field.

Reply:
Previous code creates the same `_sniAsyncAttnPacket` object over and over again. It should be created once at start (`if (_sniAsyncAttnPacket == null)`).

Reply:
Local variable `_attnPacket` has the same memory address as `_sniAsyncAttnPacket` in the line#170 in previous code. No need to create duplicated variable that has the same reference.